### ISSUE TITLE
Cannot read property 'docLimit' of undefined

### ISSUE THREAD

**Issue by Adrear** [mesqueeb/vuex-easy-firestore#159]
l like your idea and I want to play with it. But I cannot connect your plugin to my NUXT project. 
I get the same error every time. "Cannot read property 'docLimit' of undefined"

**_store/index.js_**
`import createEasyFirestore from 'vuex-easy-firestore'
const myModule = {
  firestorePath: 'userDocs/{userId}',
  firestoreRefType: 'collection', // or 'doc'
  moduleName: 'userData',
  statePropName: 'data',
  namespaced: true
}
const easyFirestore = createEasyFirestore(myModule, { logging: true })
export const plugins = [easyFirestore]
export const state = () => ({
  items: []
})
export const mutations = {
  items(state, items) { state.items = items }
}
export const getters = {
  items: state => state.items
}`
**_pages/login.vue_**
`login() {
      auth.signInWithEmailAndPassword(this.email, this.password).then((user) => {
        if (user) {
          this.$store.dispatch('userData/openDBChannel', { clearModule: true })
             .catch(console.error)
          }
        this.$router.push('/')
      }).catch((error) => {
      })
    }`

**Comment by mesqueeb** [comment]
Yes Vuex easy Firestore is not yet compatible with Nuxt and I do not have time right now to make it compatible . 
I myself have not yet used Nuxt.. 

You're free to look through the source code and make a PR! 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by eugenepentland** [comment]
@Adrear it has to do with it not reading the default values for the module. When I copied the default values in, I was able to get passed that, but then it had an error with the getters. I am working on trying to get it work with Nuxt as well.

Adding sync: { docLimit: 50} got passed that error.

```
const module = {
  firestorePath: '',
  firestoreRefType: '',
  moduleName: '',
  statePropName: '',
  logging: false,
  sync: {
    where: [],
    orderBy: [],
    fillables: [],
    guard: [],
    defaultValues: {},
    preventInitialDocInsertion: false,
    debounceTimerMs: 1000,
    insertHook: function (updateStore, doc, store) { return updateStore(doc) },
    patchHook: function (updateStore, doc, store) { return updateStore(doc) },
    deleteHook: function (updateStore, id, store) { return updateStore(id) },
    insertBatchHook: function (updateStore, docs, store) { return updateStore(docs) },
    patchBatchHook: function (updateStore, doc, ids, store) { return updateStore(doc, ids) },
    deleteBatchHook: function (updateStore, ids, store) { return updateStore(ids) },
  },
  serverChange: {
    defaultValues: {}, // depreciated
    convertTimestamps: {},
    addedHook: function (updateStore, doc, id, store) { return updateStore(doc) },
    modifiedHook: function (updateStore, doc, id, store) { return updateStore(doc) },
    removedHook: function (updateStore, doc, id, store) { return updateStore(doc) },
  },
  fetch: {
    // The max amount of documents to be fetched. Defaults to 50.
    docLimit: 50,
  }
}
}

**Comment by mesqueeb** [comment]
@eugenepentland all default values of the plugin are merged with the passed options when the plugin is triggered during vuex instantiation. 

It’s weird that you have to pass the default plugin values yourself. Does this mean that Nuxt doesn’t instantiate vuex at all?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Dinibewerbig** [comment]
Evening @Adrear , could make vuex-easy-firestore work with Nuxt in the end?

**Comment by riderx** [comment]
@eugenepentland same error here, did you find a solution ?
